File: Algorithms(Python)/Minimum Index Sum of Two Lists.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        d={}
        for i in range(len(list1)):
            if list1[i] in list2:
                d[list1[i]]=i
        for i in range(len(list2)):
            if list2[i] in d:
                d[list2[i]]+=i
        d=sorted(d.items(),key=lambda x:x[1])
        val=d[0][1]
        arr=[]
        for i in d:
            x,y=i
            if y==val:
                arr.append(x)
        return arr

File: Algorithms(Java)/Minimum Index Sum of Two Lists.java
class Solution {
    public String[] findRestaurant(String[] list1, String[] list2) {
        HashMap<String,Integer> map=new HashMap<String,Integer>();
        for(int i=0;i<list1.length;i++){
            map.put(list1[i],i);
        }
        int minimum=Integer.MAX_VALUE;
        for(int i=0;i<list2.length;i++){
            if (map.containsKey(list2[i])){
                minimum=Math.min(minimum,map.get(list2[i])+i);
                map.put(list2[i],-(map.get(list2[i])+i));
            }
        }
        List<String> l=new ArrayList<String>();
        for(Map.Entry<String,Integer> e:map.entrySet()){
            if (e.getValue()==-minimum){
                l.add(e.getKey());
            }
        }
        return l.toArray(new String[0]);
    }
}